src/test/scala/edu/ie3/simona/scheduler/SimSchedulerSpec.scala

```diff
@@ -35,248 +35,911 @@ class SimSchedulerSpec
+      doneMsg.tick shouldBe 3600L
+      doneMsg.noOfFailedPF shouldBe 0
+      doneMsg.errorInSim shouldBe false
+
+      // we expect that the scheduler answers with a SimulationSuccessfulMessage
+      // to the sender of the run request (this test)
+      expectMsg(SimulationSuccessfulMessage)
+    }
+
+    "work correctly if timeBinSize == 1 and readyCheckWindow == 1" in {
+      val triggeredAgents = Range.inclusive(1, 5).map(i => TestProbe(s"SPA_$i"))
+
+      val (simScheduler, resultEventListener) = setupScheduler(
+        timeConfig = defaultTimeConfig.copy(
+          startDateTime = "2011-01-01 00:00:00",
+          endDateTime = "2011-01-01 00:00:10",
+          schedulerReadyCheckWindow = Some(1)
+        )
+      )
+
+      triggeredAgents.foreach(actor =>
+        // send to init trigger to scheduler
+        simScheduler ! ScheduleTriggerMessage(
+          createMockInitTrigger(),
+          actor.ref
+        )
+      )
+
+      // tell scheduler to init
+      simScheduler ! InitSimMessage
+
+      resultEventListener.expectMsg(Initializing)
+
+      triggeredAgents.foreach {
+        _.expectInitAndComplete(
+          simScheduler,
+          resultEventListener,
+          Seq(1L)
+        )
+      }
+
+      // wait until init is done
+      resultEventListener.expectMsgType[InitComplete]
+
+      // tell scheduler we want to run until tick 10
+      simScheduler ! StartScheduleMessage(Some(10L))
+
+      // expect a simulating event
+      resultEventListener.expectMsg(Simulating(0L, 10L))
+
+      // interact with the scheduler for each tick
+      for (tick <- 1L to 9L) {
+        triggeredAgents.foreach {
+          _.expectAstAndComplete(
+            simScheduler,
+            resultEventListener,
+            tick,
+            Seq(tick + 1)
+          )
         }
-        pass should be
-        true
+        // we expect exactly one check window passed event as
+        // all activity start triggers for the last tick should be answered
+        resultEventListener.expectMsgType[CheckWindowPassed].tick shouldBe tick
       }
 
-    "notify with Error event on receipt of actor termination message" in {
+      // answer with completion for the last tick, schedule another tick to test if the simulation event terminates as
+      // expected
+      triggeredAgents.foreach(
+        _.expectAstAndComplete(
+          simScheduler,
+          resultEventListener,
+          10L,
+          Seq(11L)
+        )
+      )
+
+      resultEventListener.expectMsgType[CheckWindowPassed].tick shouldBe 10L
+      resultEventListener.expectMsgType[Ready].tick shouldBe 10L
+      val doneMsg = resultEventListener.expectMsgType[Done]
+      doneMsg.tick shouldBe 10L
+      doneMsg.noOfFailedPF shouldBe 0
+      doneMsg.errorInSim shouldBe false
+
+      // finally expect a simulation successful message
+      expectMsg(SimulationSuccessfulMessage)
+    }
+
+    /* exceptional cases */
```

Comment:
Do we need that if there is no exceptional case?

Reply:
I don't quite understand what you mean..?

Reply:
Nevermind i just misread the comment and thought it would belong to the test before.